internal/builtinfunctions/functions_test.go

```diff
@@ -819,3 +821,104 @@ func Test_floatToFormattedString_success(t *testing.T) {
+		map[string]any{"loop_id": 3},
+	}
+	repeatedInputs := map[string]any{
+		"a": "A", "b": "B",
+	}
+	repeatedInputName := builtinfunctions.CombinedObjPropertyConstantName
+	itemName := builtinfunctions.CombinedObjPropertyItemName
+	outputExp := []any{
+		map[string]any{itemName: items[0], repeatedInputName: repeatedInputs},
+		map[string]any{itemName: items[1], repeatedInputName: repeatedInputs},
+		map[string]any{itemName: items[2], repeatedInputName: repeatedInputs},
+	}
+	functionToTest := builtinfunctions.GetFunctions()["bindConstants"]
+	output, err := functionToTest.Call([]any{items, repeatedInputs})
+	assert.NoError(t, err)
+	assert.Equals(t, output.([]any), outputExp)
+
+	output, err = functionToTest.Call([]any{[]any{}, repeatedInputs})
+	assert.NoError(t, err)
+	assert.Equals(t, output.([]any), []any{})
+}
+
+func defaultPropertySchema(t schema.Type) *schema.PropertySchema {
+	return schema.NewPropertySchema(t, nil, false, nil, nil, nil, nil, nil)
+}
+
+func joinStrs(s1, s2 string) string {
+	return strings.Join([]string{s1, s2}, builtinfunctions.CombinedObjIDDelimiter)
+}
+
+func TestHandleTypeSchemaZip(t *testing.T) {
+	basicStringSchema := schema.NewStringSchema(nil, nil, nil)
+	basicIntSchema := schema.NewIntSchema(nil, nil, nil)
+	strTypeID := string(basicStringSchema.TypeID())
+	intTypeID := string(basicIntSchema.TypeID())
+	listInt := schema.NewListSchema(basicIntSchema, nil, nil)
+	listStr := schema.NewListSchema(basicStringSchema, nil, nil)
+
+	_, err := builtinfunctions.HandleTypeSchemaCombine(
+		[]schema.Type{basicStringSchema, basicIntSchema})
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "expected first type to be a list schema")
+
+	_, err = builtinfunctions.HandleTypeSchemaCombine(
+		[]schema.Type{listInt})
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "expected exactly two types")
+
+	myFirstObj := schema.NewObjectSchema(
+		"ObjectTitle",
+		map[string]*schema.PropertySchema{
+			"a": defaultPropertySchema(basicStringSchema),
+			"b": defaultPropertySchema(basicStringSchema),
+		})
+	listMyFirstObj := schema.NewListSchema(myFirstObj, nil, nil)
+	constantsObj := schema.NewObjectSchema(
+		"Constants",
+		map[string]*schema.PropertySchema{
+			"p_str": defaultPropertySchema(basicStringSchema),
+			"p_int": defaultPropertySchema(basicIntSchema),
+		})
+
+	type testInput struct {
+		typeArgs       []schema.Type
+		expectedResult string
+	}
+	testInputs := []testInput{
+		{
+			typeArgs:       []schema.Type{listInt, constantsObj},
+			expectedResult: joinStrs(intTypeID, constantsObj.ID()),
+		},
+		{
+			typeArgs:       []schema.Type{listMyFirstObj, constantsObj},
+			expectedResult: joinStrs(myFirstObj.ID(), constantsObj.ID()),
+		},
+		{
+			typeArgs:       []schema.Type{listMyFirstObj, basicStringSchema},
+			expectedResult: joinStrs(myFirstObj.ID(), strTypeID),
+		},
+		{
+			typeArgs:       []schema.Type{listStr, basicIntSchema},
+			expectedResult: joinStrs(strTypeID, intTypeID),
+		},
+	}
```

Comment:
Yes, what should the behavior be when the 2nd argument is a list?

What should the schema name of a `ListSchema` be? 

```golang
ListSchema.TypeID()
ListSchema.TypeID()_schemaName(ListSchema.ItemValue)
```

```golang
list
list_item
```

Reply:
When constants is a list, the entire list will end up on all items in the constants field.
I'm guessing that it will just end up as `list`, but something more detailed may be preferred.

Reply:
I _think_ @jaredoconnell is hoping for something like this:
```go
	testInputs := []testInput{
		{  // Inputs:  a list of int's and an object
			typeArgs:       []schema.Type{listInt, constantsObj},
			expectedResult: joinStrs(intTypeID, constantsObj.ID()),
		},
		{ // Inputs:  a list of objects and an object
			typeArgs:       []schema.Type{listMyFirstObj, constantsObj},
			expectedResult: joinStrs(myFirstObj.ID(), constantsObj.ID()),
		},
		{ // Inputs:  a list of objects and a string
			typeArgs:       []schema.Type{listMyFirstObj, basicStringSchema},
			expectedResult: joinStrs(myFirstObj.ID(), strTypeID),
		},
		{ // Inputs:  a list of strings and an int
			typeArgs:       []schema.Type{listStr, basicIntSchema},
			expectedResult: joinStrs(strTypeID, intTypeID),
		},
	}
```

Reply:
added the comments